### PR TITLE
Allow servers to send openable links to players

### PR DESCRIPTION
**Pull request by Phinner**
https://github.com/Anuken/Mindustry-Suggestions/issues/3666

:)

If your pull request is **not** translation or serverlist-related, read the list of requirements below and check each box:

- [x] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [x] I have ensured that my code compiles, if applicable.
- [x] I have ensured that any new features in this PR function correctly in-game, if applicable.

### CODE REVIEW

File: core/assets/bundles/bundle.properties [hunk 1]
```diff
@@ -14,6 +14,7 @@ link.f-droid.description = F-Droid listing
 link.wiki.description = Official Mindustry wiki
 link.suggestions.description = Suggest new features
 link.bug.description = Found one? Report it here
+linkopen = Are you sure to open the following link:\n[blue]{0}
```
Comment (Anuken) [inline core/assets/bundles/bundle.properties:17]:
```suggestion
linkopen = This server has sent you a link. Are you sure you want to open it?\n\n[sky]{0}
```

**Comment by LeoSko** [comment]
With no callbacks to know whether they opened a link? >-<
The filename even says **Menus**. *uh...*

**Comment by Anuken** [comment]
`Menus` has a bunch of other stuff that doesn't involve user interaction at all, and the use case for links is just convenience, I don't think the server needs to know whether it was clicked at all.

**Comment by LeoSko** [comment]
> I don't think the server

Which one of those that you host don't need interactions with users?

**Comment by Phinner** [comment]
Well, it's just sending a link, its not like a menu that has to be interactive.

**Comment by Anuken** [comment]
> Which one of those that you host don't need interactions with users?

What on earth are you talking about...? I'm referring to other functions in Menus, like `announce`, which are just static info messages. There are plenty of other ones like that, just because they're in a file called Menus doesn't mean it has to involve an actual menu.

Whether or not a user clicks a link is their business, the server should not know or care. The link is presented to them, it's their choice whether to click it. It's not a menu, it's a *suggestion*. If you want a menu, use the menu system.

**Comment by LeoSko** [comment]
> server should not know or care

If you have ever tried to create an ads campaign, or any other sort of server-side event that lasts days, you would reject this statement in no time.
Correct me if I'm wrong, but ui.showConfirm opens a blocking window with 2 options, which ehm is exactly what menu is.
If you look at the game overall, *everything* is a suggestion, nobody is obliged to do anything, hence you don't if they do or not and *what* they do or not.

**Comment by Anuken** [comment]
> Correct me if I'm wrong, but ui.showConfirm opens a blocking window with 2 options, which ehm is exactly what menu is.

What does that have to do with this function being in the Menus class, why do you bring this up at all?

> If you have ever tried to create an ads campaign, or any other sort of server-side event that lasts days, you would reject this statement in no time.

Yeah, no thanks, no "ad campaigns" in my game. Servers doing *anything* based on me clicking a link (or not) is creepy and a privacy violation. If you want to do an event, just present the link normally and let the user click if they care, no "click link for reward" nonsense.

> If you look at the game overall, everything is a suggestion, nobody is obliged to do anything, hence you don't if they do or not and what they do or not.

Bad take, obviously "everything is a choice", I could just turn off my computer. Pointless to discuss, though.

A menu option presents a choice, for which closing is an option, and the server should know about this. Links are not like this. **They are outside the game, and not a choice that the server should know about.**

**Comment by LeoSko** [comment]
> no "ad campaigns" in my game

Well, hope you'll find other ways to keep the game entertaining and players engaged with other gamemodes, for example. *blindly, as usual*

And how does this relate to privacy violation? You are already sending to server whether the user uses mobile, their viewport sizes and many other unrelated to server things. Servers **could** exist without knowing if player is using mobile or not, but they can use it optimize UI, for instance. So let's drop this support, why would anyone need that? Ah, also let's remove chat - wtf, server knows what other player typed in. This logic is absurd and makes zero sense.

Whatever, I'll make all the dirty workarounds to make it work with custom hosted redirect links. Still wonder why you want add more hussle to plugin developers.

And yeah, you've just concluded by yourself that opening a link or not is a choice, like menu choices.

**Comment by Anuken** [comment]
> Well, hope you'll find other ways to keep the game entertaining and players engaged with other gamemodes

...and *user engagement* and *entertainment* somehow requires ad campaigns? With links? In the game? I don't see that.

> Ah, also let's remove chat-

That's a nice *straw man* argument you have there, nice to see we're stooping to making stuff up.

Let me put it this way: Servers knowing whether or not I visited *a website they presented* is far worse than them knowing my viewport or other insignificant details, because it can easily be used to more or less *force* people to go to these sites, usually for rewards (or, worse, gating the player out). An *external website* knows far more about you than the game, and can even be quite dangerous. 

Oh, sure, it's still a *choice* to click, but this can be made very appealing in nasty ways, and I don't want servers to have this power over users ***at all***.

> And yeah, you've just concluded by yourself that opening a link or not is a choice, like menu choices.

I literally said they're completely different *as choices*, but, yeah, sure, conclude what you want, I suppose. Again, obviously it's *all* a choice, I could *choose* to do absolutely anything, but that doesn't make it useful to discuss, does it?

---

I'm not convinced at all, and frankly I've spent enough time on this issue as-is. No reason to continue, locking.

**Pull request merged**